examples/chip-tool/main.cpp

```diff
@@ -289,12 +294,32 @@ bool DetermineArgsBle(char * argv[], CommandArgs * commandArgs)
 
 bool DetermineArgsEcho(char * argv[], CommandArgs * commandArgs)
 {
-    if (!IPAddress::FromString(argv[2], commandArgs->hostAddr))
+    struct addrinfo hints;
+    struct addrinfo * result;
+    int ret;
+
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family   = AF_UNSPEC;
+    hints.ai_socktype = SOCK_DGRAM;
+    ret               = getaddrinfo(argv[2], nullptr, &hints, &result);
+
+    if (ret < 0)
     {
         fputs("Error: Invalid device IP address", stderr);
         return false;
     }
 
+    commandArgs->hostAddr = IPAddress::FromSockAddr(*result->ai_addr);
+    if (result->ai_family == AF_INET6)
+    {
+        struct sockaddr_in6 * addr = reinterpret_cast<struct sockaddr_in6 *>(result->ai_addr);
+        commandArgs->interfaceId   = addr->sin6_scope_id;
```

Comment:
Shouldn't this be `addressScope` not `interfaceId` ?

Reply:
The CHIP library calls it `interfaceId` and the linux doc says

```
sin6_scope_id
It is new in Linux 2.4.
       Linux supports it only for link-local addresses, in that case
       sin6_scope_id contains the interface index
```